Sources/NIOPosix/BSDSocketAPICommon.swift

```diff
@@ -127,6 +127,28 @@ extension NIOBSDSocket.Option {
         NIOBSDSocket.Option(rawValue: Posix.IPV6_RECVPKTINFO)
 }
 
+extension NIOBSDSocket {
+    struct ProtocolSubtype: RawRepresentable, Hashable {
+        typealias RawValue = CInt
+        var rawValue: RawValue
+        
+        init(rawValue: RawValue) {
+            self.rawValue = rawValue
+        }
+    }
+}
+
+extension NIOBSDSocket.ProtocolSubtype {
+    static let ip = Self(rawValue: 0)
```

Comment:
Should we add an `MPTCP` `static let` here to avoid using the raw value elsewhere?

Reply:
Done in fadd5c120b7626be22b7731daab94c171a1fe610